controller/konnect/ops/ops_konnecttransitgateway.go

```diff
@@ -0,0 +1,246 @@
+}
+
+// updateKonnectTransitGateway is called when an "Update" operation is called in reconciling a Konnect transit gateway.
+// Since Konnect does not provide API to update an existing transit gateway, here we can only update the status of the
+// KonnectCloudGatewayTransitGateway resource based on the state of the transit gateway on the Konnect side.
+func updateKonnectTransitGateway(
+	ctx context.Context,
+	sdk sdkops.TransitGatewaysSDK,
+	tg *konnectv1alpha1.KonnectCloudGatewayTransitGateway,
+) error {
+	networkID := tg.GetNetworkID()
+	if networkID == "" {
+		return CantPerformOperationWithoutNetworkIDError{
+			Entity: tg,
+			Op:     UpdateOp,
+		}
+	}
+
+	resp, err := sdk.GetTransitGateway(ctx, networkID, tg.GetKonnectID())
+	if errWrap := wrapErrIfKonnectOpFailed(err, UpdateOp, tg); errWrap != nil {
+		return errWrap
+	}
+
+	if resp == nil || resp.TransitGatewayResponse == nil {
+		return fmt.Errorf("failed updating %s: %w", tg.GetTypeName(), ErrNilResponse)
+	}
+
+	tg.Status.State = extractStateFromTransitGatewayResponse(resp.TransitGatewayResponse)
+	return nil
+}
+
+// deleteKonnectTransitGateway deletes a Konnect transit gateway.
+func deleteKonnectTransitGateway(
+	ctx context.Context,
+	sdk sdkops.TransitGatewaysSDK,
+	tg *konnectv1alpha1.KonnectCloudGatewayTransitGateway,
+) error {
+	networkID := tg.GetNetworkID()
+	if networkID == "" {
+		return CantPerformOperationWithoutNetworkIDError{
+			Entity: tg,
+			Op:     DeleteOp,
+		}
+	}
+
+	// Failures on deleting transit gateway causes `sdk.DeleteTransitGateway` stuck because the SDK performs a retry that can take up to 1 hour:
+	// https://github.com/Kong/gateway-operator/issues/1521
+	// So we disable the retry here. The reconciler retries the creation by the requeue mechanism of the controller runtime.
+	resp, err := sdk.DeleteTransitGateway(ctx, networkID, tg.GetKonnectID(), sdkkonnectops.WithRetries(retry.Config{}))
+
+	if errWrap := wrapErrIfKonnectOpFailed(err, DeleteOp, tg); errWrap != nil {
+		return errWrap
+	}
+
+	if resp == nil {
+		return fmt.Errorf("failed deleting %s: %w", tg.GetTypeName(), ErrNilResponse)
+	}
+
+	return nil
+}
+
+func getKonnectTransitGatewayMatchingSpecName(
+	ctx context.Context,
+	sdk sdkops.TransitGatewaysSDK,
+	tg *konnectv1alpha1.KonnectCloudGatewayTransitGateway,
+) (string, error) {
+	networkID := tg.GetNetworkID()
+	if networkID == "" {
+		return "", CantPerformOperationWithoutNetworkIDError{
+			Entity: tg,
+			Op:     CreateOp,
+		}
+	}
+
+	resp, err := sdk.ListTransitGateways(ctx, sdkkonnectops.ListTransitGatewaysRequest{
+		NetworkID: networkID,
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("failed listing %s: %w", tg.GetTypeName(), err)
+	}
+
+	if resp == nil || resp.ListTransitGatewaysResponse == nil {
+		return "", fmt.Errorf("failed listing %s: %w", tg.GetTypeName(), ErrNilResponse)
+	}
+	// TODO: update KonnectCloudGatewayTransitGateway to satisfy the GetID() interface to reuse getMatchingEntryFromListResponseData?
```

Comment:
👍 on defining `GetID()` to reuse `getMatchingEntryFromListResponseData` here

Reply:
Sounds good. Since it requires updates on the `KonnectCloudGatewayTransitGateway` CRD, I would like to make the change in another PR.
WDYT if we reuse the `GetKonnectID()` in the interface used in `getMatchingEntryFromListResponseData` so we do not need to change the CRD?